### ISSUE TITLE
multisession `"future_lapply"` error when R quits: Error in unserialize(node$con) : error reading from connection

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#255]
The errors are the same as HenrikBengtsson/future#143. They only appear when I quit the R session, so I cannot use `traceback()` to dig further. They affects both `"future_lapply"` parallelism on the master branch and `"future"` parallelism on the `i227-attempt2` branch (which is almost ready to alpha test). 
Targets appear to build normally anyway, and the errors do not appear with `drake` 5.0.0 (current CRAN version).

```r
library(drake)
library(future)
load_basic_example()
future::plan(multisession)
make(my_plan, parallelism = "future_lapply")
q()

##  Error in unserialize(node$con) : error reading from connection
## Calls: <Anonymous> ... doTryCatch -> recvData -> recvData.SOCKnode -> unserialize
## Execution halted
## Error in unserialize(node$con) : error reading from connection
## Calls: <Anonymous> ... doTryCatch -> recvData -> recvData.SOCKnode -> unserialize
## Execution halted
## Error in unserialize(node$con) : error reading from connection
## Calls: <Anonymous> ... doTryCatch -> recvData -> recvData.SOCKnode -> unserialize
## Execution halted
## ...
```

**Comment by wlandau** [comment]
Also, for the non-HPC backends, this one seems specific to `future::multisession`. The `sequential` and `multicore` backends appear unaffected.

**Comment by wlandau** [comment]
MWE:

```r
library(future)
future::plan(multisession)
f <- future::future(
  capture.output(1+1, type = "message")
)
q()

## Error in unserialize(node$con) : error reading from connection
## Calls: <Anonymous> ... doTryCatch -> recvData -> recvData.SOCKnode -> unserialize
## Execution halted
```

So I think it's because of [this `capture.output()` call](https://github.com/ropensci/drake/blob/2049952e181618ed011693953dd07998ff13513c/R/run.R#L24) from when I fixed #212. Not yet sure how/whether to fix it. `Drake` should continue to capture and store messages somehow so users can retrieve them with `diagnose(whatever_target)`.

**Comment by wlandau** [comment]
As @HenrikBengtsson said, users can get around this one either by using `future.callr::callr` or by creating and cleaning up an explicit PSOCK cluster. See also: https://github.com/HenrikBengtsson/Wishlist-for-R/issues/55. I think archiving this explanation is enough. I believe `drake` should still continue capturing messages for `diagnose()`.

```r
library(future)
cl <- parallel::makeCluster(2L)
plan("cluster", workers = cl)  ## same as plan(multisession, workers = 2L)
> f <- future(capture.output(message("hello"), type = "message"))
> value(f)
[1] "hello"
> f <- future(capture.output(message("world"), type = "message"))
> value(f)
[1] "world"
> parallel::stopCluster(cl)
> q("no")
```
